consumer/consumer.go

```diff
@@ -4,11 +4,14 @@
 package consumer // import "go.opentelemetry.io/collector/consumer"
 
 import (
+	"context"
 	"errors"
 
 	"go.opentelemetry.io/collector/consumer/internal"
 )
 
+type ConsumeFunc[T any] func(context.Context, T) error
```

Comment:
Are concerns from https://github.com/open-telemetry/opentelemetry-collector/issues/7051#issuecomment-1420818111 still relevant with this design?

Reply:
No, because we don't change the interface (you can see that I did not change any consumer). Also, if I understand correctly my change is backwards compatible (not breaking change) and equivalent with the previous code, except I extract the common ConsumeFunc. This only hints users that they can implement this instead of every function if their logic is signal agnostic.